sympy/simplify/cse_main.py

```diff
@@ -212,42 +212,120 @@ def _find_opts(expr):
 
     ## Process Adds and commutative Muls
 
-    def _match_common_args(Func, funcs):
+    def _match_common_args(Func, funcs, dict, F):
         if order != 'none':
             funcs = list(ordered(funcs))
         else:
             funcs = sorted(funcs, key=lambda x: len(x.args))
 
-        func_args = [set(e.args) for e in funcs]
+        # ----------------- helpers ---------------------------
+        def cfunc(dx, dy, c):
```

Comment:
I moved out the one more general routine (`pairwise_most_common`) but the other ones are helpers that are written as functions to simplify the function in which they are contained. I provided comments at the start and throughout those functions. See if it looks any better now.